docs/en/guides/81-how-databend-works/03-how-databend-json-variant-works.md

```diff
@@ -0,0 +1,314 @@
+---
+title: "How Databend JSON (Variant) Works"
+sidebar_label: "How Databend JSON Works"
+---
+
+See also:
+
+- [Variant Data Type](/sql/sql-reference/data-types/variant)
+- [Semi-Structured Functions](/sql/sql-functions/semi-structured-functions/)
+
+## Core Concepts
+
+Databend's Variant type is a flexible data type designed to handle semi-structured data like JSON. It provides Snowflake-compatible syntax and functions while delivering high performance through an efficient storage format and optimized access mechanisms. Most Variant-related functions in Databend are directly compatible with their Snowflake counterparts, making migration seamless for users familiar with Snowflake's JSON handling capabilities.
+
+```
+┌─────────────────────────────────────────────────────────────────┐
+│ Variant Type Core Components                                    │
+├─────────────────┬───────────────────────────────────────────────┤
+│ Storage Format  │ JSONB-based binary storage                    │
+│ Virtual Columns │ Automatic extraction of JSON paths            │
```

Comment:
virtual column is currently an experimental feature and is not enabled by default. It can only be used after being enabled through `set enable experimental virtual column=1`